rotkehlchen/db/dbhandler.py

```diff
@@ -405,62 +427,63 @@ def get_settings(self) -> DBSettings:
         )
         query = query.fetchall()
 
-        settings: DBSettings = {}
+        settings = DBSettings()
+
         for q in query:
             if q[0] == 'version':
-                settings['db_version'] = int(q[1])
+                settings.db_versions = int(q[1])
             elif q[0] == 'last_write_ts':
-                settings['last_write_ts'] = int(q[1])
+                settings.last_write_ts = int(q[1])
             elif q[0] == 'premium_should_sync':
-                settings['premium_should_sync'] = str_to_bool(q[1])
+                settings.premium_should_sync = str_to_bool(q[1])
             elif q[0] == 'include_crypto2crypto':
-                settings['include_crypto2crypto'] = str_to_bool(q[1])
+                settings.include_crypto2crypto = str_to_bool(q[1])
             elif q[0] == 'anonymized_logs':
-                settings['anonymized_logs'] = str_to_bool(q[1])
+                settings.anonymized_logs = str_to_bool(q[1])
             elif q[0] == 'last_data_upload_ts':
-                settings['last_data_upload_ts'] = int(q[1])
+                settings.last_data_upload_ts = int(q[1])
             elif q[0] == 'ui_floating_precision':
-                settings['ui_floating_precision'] = int(q[1])
+                settings.ui_floating_precision = int(q[1])
             elif q[0] == 'taxfree_after_period':
-                settings['taxfree_after_period'] = int(q[1]) if q[1] else None
+                settings.taxfree_after_period = int(q[1]) if q[1] else None
             elif q[0] == 'balance_save_frequency':
-                settings['balance_save_frequency'] = int(q[1])
+                settings.balance_save_frequency = int(q[1])
             elif q[0] == 'include_gas_costs':
-                settings['include_gas_costs'] = str_to_bool(q[1])
+                settings.include_gas_costs = str_to_bool(q[1])
             else:
-                settings[q[0]] = q[1]
+                settings.other = q[1]
```

Comment:
I guess here this can't work. We should no longer have this else. All settings should be spelled out and if we get here this else should raise an exception.

Can you raise an AssertionError here?

`raise AssertionError(f'Unknown setting {q[0]} found in the database')`

In the future we may turn this into a more specific error but for now this should be enough.